tasks/tasks.go

```diff
@@ -0,0 +1,93 @@
+package tasks
+
+import (
+	"errors"
+	"math/rand"
+	"sync"
+	"time"
+)
+
+// Status is the status of the task.
+type Status int
+
+// TaskExecutor executes a series of task functions.
+type TaskExecutor interface {
+	QueueTask(task func() error) string
+	GetTaskStatus(id string) (Status, error)
+	completeTask(id string, task func() error)
+}
+
+type defaultExecuter struct {
+	sync.RWMutex
+	m map[string]Status
+}
+
+// These are some enumerated Status constants.
+// INPROGRESS: Task is still in progress.
+// SUCCESS: Task finished successfully.
+// FAILURE: Task finished unsuccessfully.
+// NOTFOUND: Task could not be found.
+const (
+	INPROGRESS Status = iota
+	SUCCESS
+	FAILURE
+	NOTFOUND
+)
+
+// NewTaskExectuer returns a TaskExecutor ready to execute.
+func NewTaskExectuer() TaskExecutor {
+	return &defaultExecuter{m: make(map[string]Status)}
+}
+
+// QueueTask initializes a new task. It takes a generic task function.
```

Comment:
I like your wording better, but for the record doc comments can be multiple sentences.